app/models/mortgage_calculator/tax_calculator.rb

```diff
@@ -0,0 +1,87 @@
+module MortgageCalculator
+  # base class and common logic for tax calculators
+  class TaxCalculator
+    include ActiveModel::Validations
+    include ActiveModel::Conversion
+    include CurrencyInput::Macro
+    include ActionView::Helpers::NumberHelper
+
+    attr_reader :price
+    attr_accessor :buyer_type
+    currency_inputs :price
+    validates :price, presence: true, numericality: true
+
+    SECOND_HOME_THRESHOLD = 40_000
+    SECOND_HOME_ADDITIONAL_TAX = 3
+
+    def initialize(options = {})
+      self.price = options.fetch(:price) { 0 }
+      self.buyer_type = options[:buyer_type] || 'isNextHome'.freeze
+    end
+
+    %i[price tax_due total_due].each do |field|
+      define_method "#{field}_formatted" do
+        number_to_currency(
+          public_send(field).presence || 0, unit: '', precision: 0
+        )
+      end
+    end
+
+    def tax_due
+      self.class.banding_for(bands_to_use).map do |band|
+        tax_for_band(band[:start], band[:end], band[:rate])
+      end.inject(:+)
+    end
+
+    def total_due
+      price + tax_due
+    end
+
+    def persisted?
+      false
+    end
+
+    def percentage_tax
+      result = (tax_due / price) * 100
+      result.nan? ? 0 : result
```

Comment:
Does `result.to_i` works instead?

Reply:
Much of this is just a direct lift from the old stamp duty calculator so yeah probably :)

Reply:
These are BigDecimal numbers so what can happen is if you get a divide by 0 you can get a NaN and you cant `to_i` this.